src/libraries/Microsoft.Extensions.Http/src/DefaultHttpClientFactory.cs

```diff
@@ -233,11 +245,44 @@ internal virtual void StopCleanupTimer()
         {
             lock (_cleanupTimerLock)
             {
-                _cleanupTimer!.Dispose();
+                _cleanupTimer?.Dispose();
                 _cleanupTimer = null;
             }
         }
 
+        // protected for tests
+        protected bool _disposed;
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+            _disposed = true;
+
+            StopCleanupTimer();
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        ~DefaultHttpClientFactory()
+        {
+            Dispose(false);
+        }
+
+        private void ThrowIfDisposed()
+        {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException(nameof(DefaultHttpClientFactory));
+            }
```

Comment:
Use `ObjectDisposedException.ThrowIf(_disposed, this)` for `NET7_0_OR_GREATER`?

Reply:
> I am not sure how this will help?

The exception helpers are there to reduce the size of method IL. Take a look at the technical details here: https://learn.microsoft.com/en-us/dotnet/communitytoolkit/diagnostics/throwhelper#technical-details

Reply:
I would just move the `throw new ObjectDisposedException()` to its own method so that the if check is more inlineable.
```C#
private void ThrowIfDisposed()
{
    if (_disposed)
    {
        ThrowDisposed();
        static void ThrowDisposed() => throw new ObjectDisposedException(nameof(DefaultHttpClientFactory));
    }
}
```
It's not worth complicating the code further here with ifdefs. And given where ThrowIfDisposed is used, even this minor tweak is borderline.